faker.go

```diff
@@ -711,7 +708,223 @@ func getValue(a interface{}, seenPointers *int, nilIndex int) (reflect.Value, er
 		err := fmt.Errorf("no support for kind %+v", t)
 		return reflect.Value{}, err
 	}
+}
+
+// seenPointers is used to keep track of the number of pointers that have been encountered
+// while recursing. A pointer will be set to nil if seenPointers matches nilIndex. If nilIndex is -1,
+// no pointers will be set to nil.
+func getValueSkipFields(a interface{}, seenPointers *int, nilIndex int, skipMap map[string]struct{}) (reflect.Value, error) {
```

Comment:
To reduce code duplication of this fairly complicated function, can `getValue` maybe be implemented as a call to `getValueSkipFields` with an empty `skipMap` map?

Reply:
It is a good idea